cultivation/scripts/flashcore/yaml_processor.py

```
@@ -1,426 +1,314 @@
-                        return YAMLProcessingError(
-                            file_path=source_file_path, card_index=card_index, card_question_snippet=card_q_preview,
-                            field_name="media", message=f"Media file not found at expected path: '{(assets_root_directory / media_path)}'."
+    def _process_single_yaml_file(
+        self, file_path: Path
+    ) -> Tuple[List[Card], List[YAMLProcessingError]]:
+        """Processes a single YAML file."""
+        try:
+            content = file_path.read_text(encoding="utf-8")
+            raw_yaml_content = yaml.safe_load(content)
+        except FileNotFoundError:
+            raise YAMLProcessingError(file_path, "File not found.")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)